presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -128,11 +128,11 @@ public ConnectorTableMetadata getTableMetadata(ConnectorSession session, Connect
     {
         ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> columns = ImmutableMap.builder();
         List<SchemaTableName> tables;
```

Comment:
This could be done as
```java
List<SchemaTableName> tables = prefix.getTable()
        .map(ignored -> ImmutableList.of(prefix.toSchemaTableName()))
        .orElseGet(() -> listTables(session, prefix.getSchema());
```
Though the `ignored` part is ugly. We could add method and do it like
```java
List<SchemaTableName> tables = prefix.optionalSchemaTableName()
        .map(ImmutableList::of)
        .orElse(() -> listTables(session, prefix.getSchema());
```
Just a thought, feel free to ignore this suggestion.

Reply:
Thinking about this a bit more, having `optionalSchemaTableName()` is nice because `toSchemaTableName()` can throw, same as `Optional.get()`.

Reply:
Ah, I didn't realize about the generics problem. That's always annoying.